graphql_compiler/compiler/workarounds/orientdb_class_with_while.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2017 Kensho Technologies, Inc.
+"""Workarounds for OrientDB bug disallowing "class:" clauses together with "while:" clauses.
+
+For details, see:
+https://github.com/orientechnologies/orientdb/issues/8129
+"""
+import funcy
+
+from ..blocks import Filter, Recurse
+from ..expressions import BinaryComposition, Literal, LocalField
+
+
+def _coerce_block_to_filter_block(coerce_type_block, where_block):
+    """Create an "INSTANCEOF" Filter block from a CoerceType block."""
+    coerce_type_target = coerce_type_block.target_class
+    if len(coerce_type_target) != 1:
+        raise AssertionError(u'Unexpected "coerce_type_target" for MATCH query: '
+                             u'{}'.format(coerce_type_target))
+    coerce_type_target = funcy.first(coerce_type_target)
```

Comment:
The length check above is strict. You can index[0] element here to ensure that you don't end up with a Maybe data type

Reply:
+1 it would also read cleaner

Reply:
`coerce_type_target` is unfortunately a `set` so `[0]` isn't an option. The length check ensures that the `Maybe` is always a `Some` (or in python3 terms, the `Optional[str]` is actually just a `str`).